internal/testutils/jimmtest/env.go

```
@@ -183,9 +183,26 @@ func (m Model) addModelRelations(c *qt.C, db *db.Database, client *openfga.OFGAC
 }
 
 // addControllerRelations adds permissions the model should have and adds permissions for users to the controller.
-func (ctl Controller) addControllerRelations(c *qt.C, client *openfga.OFGAClient) {
+func (ctl Controller) addControllerRelations(c *qt.C, db *db.Database, client *openfga.OFGAClient) {
 	err := client.AddCloudController(context.Background(), names.NewCloudTag(ctl.Cloud), ctl.dbo.ResourceTag())
 	c.Assert(err, qt.IsNil)
+	for _, u := range ctl.Users {
+		dbUser := ctl.env.User(u.User).DBObject(c, db)
+		var relation openfga.Relation
+		switch u.Access {
+		case "admin":
+			relation = ofganames.AdministratorRelation
+		case "add-model":
+			relation = ofganames.CanAddModelRelation
+		default:
+			c.Fatalf("unknown controller access level: %s", u.Access)
+		}
+		if client != nil {
```

We just used the client on line 187, also as it's just for tests it's ok to let it panic.

---

Actually wouldn't a simple lookup map look a bit better here

```go
relation, ok := map[string]openfga.Relation{
    "admin":     ofganames.AdministratorRelation,
    "add-model": ofganames.CanAddModelRelation,
}[u.Access]
if !ok {
    c.Fatalf("unknown controller access level: %s", u.Access)
}
dbUser := ctl.env.User(u.User).DBObject(c, db)
// ... so forth
```

Could also just remove the ok check and let the SetControllerAccess fail with "" or whatever it is.

---

This was just c/p, removed the `if client != nil` check but left the rest the same.

---

Ah sure, ty for removing nil check.